### DISCUSSION TITLE
Could we reuse zod input schemas made just for parsing/validation?

### DISCUSSION THREAD

**Discussion by lkj4** [trpc/trpc#2120] [Ideas]
And this without the need to declare a dedicated zod schema outside my RPC definition (just to have less boilerplate and naming fatigue).

My use case is to narrow down `unknown` input types without using `InferQueryInput<'hello'>` because I'd like to avoid the `'hello'` string which can't be auto-refactored with F2. Is this possible or easy to implement?

**Comment by KATT** [comment]
I think you want `z.input<typeof schema>` or `z.output<typeof schema>`

This code might give you some ideas: https://kitchen-sink.trpc.io/react-hook-form?file=feature%2Freact-hook-form%2Findex.tsx#content
